chore: merge cloud-cli to main

cjohnhanson opened the pull request:
Opening a draft PR for the big cloud->main merge to track conflicts and changes ahead of GA launch.

WillDaSilva commented:
@cjohnhanson Instead of using a PR, which will cause all of the commits to be squashed thereby losing the history of all of the PRs we've merged into `cloud`, how do you feel about us just force-pushing `cloud` to `main`, thereby essentially switching the main branch over to the commits we've pushed to `cloud`. After that we can update our docs to remove references to the `cloud` branch, and eventually delete the `cloud` branch once we're pretty sure no one is still using it (e.g. in a year or so).

In either case I think we'll want to manually write the changelog for the Meltano v3 release, rather than relying on the normal auto-generated changelog.

edgarrmondragon commented:
@WillDaSilva I think we could also enable rebase or merge commits temporarily and merge this PR with either option

edgarrmondragon committed:
fix(cloud): Validate ULID inputs (#8055)

WillDaSilva committed:
fix: Permit selection of project by unambiguous name (#8060)

Closes #8057

Commit:
feat: Cloud CLI for adding projects (#7620)

* wip

* first pass

* await response

* tests and pr feedback

* tidy up

* Update src/cloud-cli/meltano/cloud/cli/project.py

Co-authored-by: Edgar R. M. <edgar@meltano.com>

* Update src/cloud-cli/tests/cli/test_project.py

Co-authored-by: Edgar R. M. <edgar@meltano.com>

* poetry update

* Project add command

* Use new provisioner API projects endpoint in project add command

* Use yaspin

* [pre-commit.ci] auto fixes from pre-commit.ci hooks

for more information, see https://pre-commit.ci

* Add -> create

* Add -> create in tests

* [pre-commit.ci] auto fixes from pre-commit.ci hooks

for more information, see https://pre-commit.ci

* feat: Deploy project stack on creation (#7925)

* Project add command

* Use new provisioner API projects endpoint in project add command

* Use yaspin

* [pre-commit.ci] auto fixes from pre-commit.ci hooks

for more information, see https://pre-commit.ci

* Add -> create

* Add -> create in tests

* [pre-commit.ci] auto fixes from pre-commit.ci hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

* Update src/cloud-cli/meltano/cloud/cli/project.py

Co-authored-by: Will Da Silva <will@willdasilva.xyz>

* Update src/cloud-cli/meltano/cloud/cli/project.py

Co-authored-by: Will Da Silva <will@willdasilva.xyz>

* Update src/cloud-cli/meltano/cloud/cli/project.py

Co-authored-by: Will Da Silva <will@willdasilva.xyz>

* Update src/cloud-cli/meltano/cloud/cli/project.py

Co-authored-by: Will Da Silva <will@willdasilva.xyz>

* Update src/cloud-cli/tests/cli/test_project.py

Co-authored-by: Will Da Silva <will@willdasilva.xyz>

* Update src/cloud-cli/tests/cli/test_project.py

Co-authored-by: Will Da Silva <will@willdasilva.xyz>

* Don't print created project json

* project_name -> name

* repository-url -> git-repository

* correct project create args in test

* correct project create args in test

* project-root-path -> root-path

* [pre-commit.ci] auto fixes from pre-commit.ci hooks

for more information, see https://pre-commit.ci

* Update project create test

* Update src/meltano/cloud/cli/project.py

Co-authored-by: Edgar R. M. <edgar@meltano.com>

* Update src/meltano/cloud/cli/project.py

Co-authored-by: Edgar R. M. <edgar@meltano.com>

* Update src/meltano/cloud/cli/project.py

Co-authored-by: Edgar R. M. <edgar@meltano.com>

* PR feedback

* Add project create to cloud-cli docs.

* More flexible expected stdout output for project create test

* ignore type checks for yaspin

* [pre-commit.ci] auto fixes from pre-commit.ci hooks

for more information, see https://pre-commit.ci

* [pre-commit.ci] auto fixes from pre-commit.ci hooks

for more information, see https://pre-commit.ci

* Addressing pr feedback for docs.

---------

Co-authored-by: Edgar R. M. <edgar@meltano.com>
Co-authored-by: Cody J. Hanson <cody@meltano.com>
Co-authored-by: Cody J. Hanson <cody@codyjhanson.com>
Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
Co-authored-by: Will Da Silva <will@willdasilva.xyz>

Commit:
feat(cloud): Add cloud notifications cli for list, set, update and delete notifications (#7968)

* Add notifications cli for set, update and delete

* [pre-commit.ci] auto fixes from pre-commit.ci hooks

for more information, see https://pre-commit.ci

* Fix set

* [pre-commit.ci] auto fixes from pre-commit.ci hooks

for more information, see https://pre-commit.ci

* Better error messaging and docs

* add list and modify update endpoint for new

* finish up documentation for notification

* taylor review comments, improve docs

* remove filters from list

* --amend

* Add event payload docs

* review comments

* display filters, review comments taylor

* fix empty error message, and pre commit issues

* Fix mypy errors

* Fix pre commit issues

* [pre-commit.ci] auto fixes from pre-commit.ci hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

WillDaSilva committed:
Merge branch 'main' into cloud

WillDaSilva committed:
Re-add mistakenly removed files

Commit:
fix: Re-add cloud-cli script (#8104)

Re-add cloud-cli script

Commit:
feat: Add meltano cloud "state" command (#8070)

* WIP

* Added commands

* Added tests

* [pre-commit.ci] auto fixes from pre-commit.ci hooks

for more information, see https://pre-commit.ci

* noqa for cli imports

* type ignore for requests

* type ignore

* type ignore

* Add oneshot request matcher for state set

* feat: Add docs for new cloud state command. (#8073)

* Add docs for new cloud state command.

* Add `sh` for code blocks.

* Add help text for all commands

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

edgarrmondragon committed:
feat(cloud): Display project root path in projects list (#8122)
